### PR TITLE
Added all-your-base exercise to JavaScript track

### PR DESCRIPTION
**Pull request by javaeeeee**
Added all-your-base exercise to the JavaScript track

### CODE REVIEW

File: exercises/all-your-base/all-your-base.spec.js [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+const Converter = require('./all-your-base');
+
+const converter = new Converter();
+
+describe('Converter', function () {
+
+  it('wrong input_base base less than two', function () {
+    expect(function () {
+      converter.convert([0]);
```
Comment (rchavarria) [inline exercises/all-your-base/all-your-base.spec.js]: To reflect better what line 9 describes, shouldn't this line be:

    converter.convert([0], 1);

Reading the test, it looks like `[0]` is the `inputBase` parameter, but in fact, it's `array` parameter.

What do you think? Do you think users will understand better this test adding the `inputBase` parameter as `1` (or another value less than `2`)?

File: exercises/all-your-base/all-your-base.spec.js [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+
+  xit('wrong input_base base not integer', function () {
+    expect(function () {
+      converter.convert([0], 2.5);
+    }).toThrow(new Error('Wrong input base'));
+  });
+
+  xit('wrong output_base base less than two', function () {
+    expect(function () {
+      converter.convert([0], 2);
```
Comment (rchavarria) [inline exercises/all-your-base/all-your-base.spec.js]: Same as comment on line 11

**Comment by rchavarria** [comment]
Awesome!

It's nice to have a new exercise in this track.

After talking about comments on lines 11 and 23, would you mind to squash your commits? Would you need some help for doing that?

**Comment by javaeeeee** [comment]
Hello Ruben,

Thank you for reviewing the code. I would like to mention that the tests
you commented on were intended to test the cases of missing bases as
JavaScript allows to pass as little arguments as you like. :)

There are separate tests that test unity as a base like 'first base is one'
and 'second base is one' on lines 99 and 105.

If you thank that testing the case of missing arguments is redundant, I can
remove them altogether. Or maybe, it's better to change the description of
the tests?

Thank you.

**Comment by rchavarria** [comment]
You can keep tests without the parameter that are currently on L9 and L21. They show a specific feature of JavaScript. But, I would move the close to the ones you mentioned on L99 and L105 (I didn't notice them). And maybe, a better description could be `missing input/output base throws an error` or similar.

**Comment by javaeeeee** [comment]
Hello Ruben,

Thank you for your comments. Renamed and moved the tests checking missing
inputs. Also squashed the commits.

**Comment by rchavarria** [comment]
Thanks so much for this new exercise.

**Comment by rchavarria** [comment]
I'm going to close this PR manually. I'm not sure why it wasn't marked as *merged* automatically when I pushed the merge :cry:

**Comment by javaeeeee** [comment]
Thank you.

**Pull request closed without merging**